torch/_inductor/kernel/flex/templates/flex_decode.py.jinja

```diff
@@ -120,6 +111,19 @@
     # Offset the kv_indices tensor by the correct batch and head
     kv_indices = KV_IDX + sparse_idx_hz_offset
     kv_num_blocks = tl.load(KV_NUM_BLKS + sparse_block_hz_offset)
+
+    # Make sure each split is a multiple of BLOCK_N
+    if HAS_FULL_BLOCKS:
+        TILE_KV_OG = tl.cdiv(KV_LEN, SPLIT_KV)
```

Comment:
why this for full_blocks?

Reply:
There is no fundamental reason that the optimization can not be applied to full block. But we don't have a tests covering that due to this setting in vllm: https://github.com/vllm-project/vllm/blob/61249b177de1566027fc74e9b51b45a4c973eb47/vllm/v1/attention/backends/flex_attention.py#L615-L616 .

I think it would be good to revisit if one day vllm can pass in full blocks